tests/testthat/test_parse_obs_types.R

```diff
@@ -0,0 +1,46 @@
+test_that("single observation type, scale not supplied", {
+  obs1a <- list(
+    variable = "CONC"
+  )
+  expect_equal(
+    parse_obs_types(obs1a),
+    "      obs.insert(obs.end(), CONC/(1.0)); "
+  )
+})
+
+test_that("single observation type, scale=variable", {
+  obs1b <- list(
+    variable = "CONC",
+    scale = "V/1000.0"
+  )
+  expect_equal(
+    parse_obs_types(obs1b),
+    "      obs.insert(obs.end(), CONC/(V/1000.0)); "
+  )
+})
+
+test_that("multiple observation type, scale is single", {
+  obs3a <- list(
+    variable = c("CONC", "CONCM", "PD"),
+    scale = 1
+  )
+  expect_warning(
+    res <- PKPDsim:::parse_obs_types(obs3a)
+  )
```

Comment:
```suggestion
  expect_warning(
    res <- parse_obs_types(obs3a),
    "Provided `scale` vector not same length as `variable` vector."
  )
```
if i'm correct this is a warning we are raising? might want to check for it